instrumentation/java-httpclient/src/main/java/io/opentelemetry/instrumentation/auto/httpclient/JdkHttpClientTracer.java

```diff
@@ -66,10 +70,22 @@ protected String requestHeader(HttpRequest httpRequest, String name) {
   }
 
   @Override
-  protected String responseHeader(HttpResponse httpResponse, String name) {
+  protected String responseHeader(HttpResponse<?> httpResponse, String name) {
     return httpResponse.headers().firstValue(name).orElse(null);
   }
 
+  @Override
+  protected Span onResponse(Span span, HttpResponse<?> httpResponse) {
+    span = super.onResponse(span, httpResponse);
+
+    if (httpResponse != null) {
+      SemanticAttributes.HTTP_FLAVOR.set(
+          span, httpResponse.version() == Version.HTTP_1_1 ? "1.1" : "2.0");
```

Comment:
interesting, makes sense for a negotiated connection, where we don't know flavor ahead of time